app/src/main/java/com/github/se/eduverse/ui/archive/ArchiveScreen.kt

```diff
@@ -0,0 +1,77 @@
+package com.github.se.eduverse.ui.archive
+
+import androidx.compose.foundation.clickable
+import androidx.compose.foundation.layout.Column
+import androidx.compose.foundation.layout.fillMaxSize
+import androidx.compose.foundation.layout.fillMaxWidth
+import androidx.compose.foundation.layout.padding
+import androidx.compose.material.Card
+import androidx.compose.material.icons.Icons
+import androidx.compose.material.icons.automirrored.filled.ArrowBack
+import androidx.compose.material3.CenterAlignedTopAppBar
+import androidx.compose.material3.ExperimentalMaterial3Api
+import androidx.compose.material3.Icon
+import androidx.compose.material3.IconButton
+import androidx.compose.material3.Scaffold
+import androidx.compose.material3.Text
+import androidx.compose.material3.TopAppBarDefaults
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.collectAsState
+import androidx.compose.runtime.getValue
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.platform.testTag
+import androidx.compose.ui.unit.dp
+import com.github.se.eduverse.model.Folder
+import com.github.se.eduverse.ui.navigation.BottomNavigationMenu
+import com.github.se.eduverse.ui.navigation.LIST_TOP_LEVEL_DESTINATION
+import com.github.se.eduverse.ui.navigation.NavigationActions
+import com.github.se.eduverse.ui.navigation.Route
+import com.github.se.eduverse.viewmodel.FolderViewModel
+
+@OptIn(ExperimentalMaterial3Api::class)
+@Composable
+fun ArchiveScreen(navigationActions: NavigationActions, folderViewModel: FolderViewModel) {
+  folderViewModel.getArchivedUserFolders()
```

Comment:
The folderViewModel.getArchivedUserFolders() method is called directly in the body of the ArchiveScreen function. If this function is called multiple times (for example, when recomposing the screen), it could cause repeated requests. One could be to move this call into the initialization of the ViewModel or use a mechanism like LaunchedEffect.